client-spark/common/src/main/java/org/apache/spark/shuffle/ShuffleHandleInfo.java

```diff
@@ -97,21 +105,83 @@ public ShuffleServerInfo useExistingReassignmentForMultiPartitions(
   public ShuffleServerInfo createNewReassignmentForMultiPartitions(
       Set<Integer> partitionIds, String faultyServerId, ShuffleServerInfo replacement) {
     if (replacement != null) {
-      faultyServerReplacements.put(faultyServerId, replacement);
+      faultyServerToReplacements.put(faultyServerId, replacement);
     }
 
-    replacement = faultyServerReplacements.get(faultyServerId);
+    replacement = faultyServerToReplacements.get(faultyServerId);
     for (Integer partitionId : partitionIds) {
       List<ShuffleServerInfo> replicaServers = partitionToServers.get(partitionId);
       for (int i = 0; i < replicaServers.size(); i++) {
         if (replicaServers.get(i).getId().equals(faultyServerId)) {
```

Comment:
There a little problem, when reassign server for the second time, faultyServerId is not in partitionToServers.

Reply:
Currently, the server reassignment will only occur once time.